Source/Mosa.Compiler.MosaTypeSystem/Units/GenericArgumentsCollection.cs

```diff
@@ -7,7 +7,7 @@ namespace Mosa.Compiler.MosaTypeSystem
 	/// <summary>
 	/// Removes duplicates and preserves order
 	/// </summary>
-	public class GenericArgumentsCollection : List<MosaType>
+	public class GenericArgumentsCollection : List<MosaType?>
```

Comment:
Is it valid to have nulls in this list?

Reply:
Theoretically speaking an argument's actual type cannot be null, however removing this null check will cause warnings, due to other potentially-null MosaType instances being added into the collection.

Reply:
If it shouldn't have null then don't allow nulls, if there are warnings when trying to add to this list then doing a null check and throwing an error would be a better solution.

Reply:
Yeah that's probably a better idea. Will fix soon!